allennlp/nn/util.py

```diff
@@ -567,3 +567,42 @@ def logsumexp(tensor: torch.Tensor,
     else:
         stable_vec = tensor - max_score.unsqueeze(dim)
     return max_score + (stable_vec.exp().sum(dim, keepdim=keepdim)).log()
+
+
+def add_bos_eos(tensor, mask, bos, eos):
+    """
+    Add begin/end of sentence tokens to the batch of sentences.
+    Given a batch of sentences with size ``(batch_size, n_times)`` or
+    ``(batch_size, n_times, dim)`` this returns a tensor of shape ``(batch_size, n_times + 2)`` or
+    ``(batch_size, n_times + 2, dim)`` respectively.
+
+    Returns both the new tensor and updated mask
+
+    Parameters
+    ----------
+    tensor : A tensor of shape ``(batch_size, n_times)`` or ``(batch_size, n_times, dim)``
```

Comment:
`timesteps` rather than `n_times`, here and above.

Reply:
done